components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/CanBeSubclassDeterminer.scala

```diff
@@ -156,10 +144,47 @@ trait CanBeSubclassDeterminer {
       ) orElse
         isAssignable(givenClass.klass, superclassCandidate.klass)
 
-    val canBeSubclass = equalClassesOrCanAssign andThen hasSameTypeParams
+    val canBeSubclass = equalClassesOrCanAssign andThen (_ => typeParametersMatches(givenClass, superclassCandidate))
     canBeSubclass orElse canBeConvertedTo(givenType, superclassCandidate)
   }
 
+  private def typeParametersMatches(givenClass: TypedClass, superclassCandidate: TypedClass) = {
+    def canBeSubOrSuperclass(givenClassParam: TypingResult, superclassParam: TypingResult) =
```

Comment:
if we were to introduce naming like `isCovariant` maybe this method could be called something like `isSubtypeConsistent` or something like this? and the code could be something like:
```scala
givenClassParam.isCovariant(superclassParam) || givenClassParam.isContravariant(superclassParam)
```
WDYT?